tests/unittests/utils/test_qwen3_8.py

```
@@ -0,0 +1,355 @@
+    @pytest.mark.parametrize("effort", ["high", "none", "XHIGH", ""])
+    def test_rejects_unsupported_reasoning_effort(self, effort):
+        """The chat template raises on anything outside xhigh/medium/low,
+        so the config rejects it before a generation is attempted"""
+        with pytest.raises(ValueError, match="reasoning_effort"):
+            Qwen38ModelConfig({"reasoning_effort": effort})
+
+
+class TestQwen38PrepareImage:
+    """Test image normalization to PIL, which needs no model"""
+
+    def _prepare(self, img):
```

Ruff (0.16.1)

[warning] Missing return type annotation for private function `_prepare`

(ANN202)